neuraxle/pipeline.py

```diff
@@ -273,92 +120,153 @@ def fit(self, data_inputs, expected_outputs=None) -> 'Pipeline':
             expected_outputs=expected_outputs
         )
 
-        new_self, _ = self._fit_transform_core(data_container)
+        context = ExecutionContext.from_root(self, self.cache_folder)
+
+        new_self, data_container = self._fit_core(data_container, context)
 
         return new_self
 
-    def inverse_transform_processed_outputs(self, processed_outputs) -> Any:
+    def inverse_transform(self, processed_outputs) -> Any:
         """
         After transforming all data inputs, and obtaining a prediction, we can inverse transform the processed outputs
 
         :param processed_outputs: the forward transformed data input
         :return: backward transformed processed outputs
         """
         for step_name, step in list(reversed(self.items())):
-            processed_outputs = step.transform(processed_outputs)
+            processed_outputs = step.inverse_transform(processed_outputs)
```

Comment:
Could `for step_name, step in reversed(self)` do the trick? Does it apply mutates already? So reversed(self) creates a truncable step with reversed items for which inverse_transforms were applied to transforms, and the fact that we iterate on the truncable step with __iter__ returns the `step_name, step` tuple such that the for loop works.

Reply:
Anyway, your code looks like it would work, seems fine.